### ISSUE TITLE
Missing event api's for signals 

### ISSUE THREAD

**Issue by igdianov** [Activiti/Activiti#2272]
It seems there are missing events in activiti-api that should be produced when signal is emitted  received by the subscribed consumer.

**Comment by salaboy** [comment]
You mean, the message based approach for this: https://github.com/Activiti/activiti-api/blob/develop/activiti-api-process-runtime/src/main/java/org/activiti/api/process/runtime/ProcessRuntime.java#L115 ? or the events that are generated by that operation?

**Comment by igdianov** [comment]
@salaboy I mean the event are not generated for signal operations...

**Comment by CTI777** [comment]
@igdianov @salaboy 
Can somebody advise me what is not working exactly, maybe an existing test to start with?
I have seen that signals are handled, in general, differently compared to other process, tasks, variables events. In audit service I see no traces of signal at all...

**Comment by salaboy** [comment]
@CTI777 the problem is that in the Audit Service we don't have a handler for Signal Events, so when the Runtime Bundle creates a signal event, we just ignore this in Audit. We just need to create a new SignalEvent handler in Audit to store the event as is in the database.

**Comment by CTI777** [comment]
@salaboy So the problem is only audit! And there are no problems in RB, etc....

**Comment by salaboy** [comment]
only on Audit

**Comment by CTI777** [comment]
@salaboy 
I have analyzed the issue again and am sure that this issue requires many changes for the Signal event: from activiti-api, cloud-activiti-api, activiti, converters... We do not have this event, do not have any delegates or converters for this event...
ActivitiEventType has ACTIVITY_SIGNALED, ACTIVITY_MESSAGE_RECEIVED, ACTIVITY_ERROR_RECEIVED which have relation to signal...
Maybe we can discuss this once in details...

**Comment by salaboy** [comment]
@CTI777 let's do a quick zoom today.. ping me when you are around

**Comment by salaboy** [comment]
You are right.. we don't even have the listener here: https://github.com/Activiti/Activiti/tree/develop/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/event/internal

That should be our first step.

**Comment by CTI777** [comment]
@salaboy 
I think we need to define SignalEvent in activiti-api.
Should it be defined in model shared part, like below: 
https://github.com/Activiti/activiti-api/tree/develop/activiti-api-model-shared/src/main/java/org/activiti/api/model/shared/event
https://github.com/Activiti/activiti-api/tree/develop/activiti-api-model-shared-impl/src/main/java/org/activiti/api/runtime/event/impl
Or in process runtime part, like here:
https://github.com/Activiti/activiti-api/tree/develop/activiti-api-process-runtime/src/main/java/org/activiti/api/process/runtime/events

**Comment by salaboy** [comment]
@CTI777 I don't think that we have enough time to do this before the release. 
Let's move this one to 1.1

**Comment by CTI777** [comment]
Following modules should be updated:

- [ ]  Activiti/activiti-api Implementation of ACTIVITY_SIGNALED event Activiti/activiti-api#84

- [ ] Activiti/activiti-cloud-api  Implementation of ACTIVITY_SIGNALED event Activiti/activiti-cloud-api#65

- [ ] Activiti/activiti    Implementation of ACTIVITY_SIGNALED event #2511

- [ ] Activiti/activiti-cloud-runtime-bundle-service   in progress

- [ ] Activiti/activiti-cloud-audit-service   in progress
